R/bugzilla.R

```diff
@@ -95,133 +171,159 @@ download_bugzilla_rest_issues <- function(bugzilla_site, start_timestamp, save_f
-#' Thus, the max_bugs parameter needs to be set correctly to ensure all bugs are retrieved and that
-#' the json data is not broken. If you get an error trying to parse the data downloaded with this
-#'
-#' @param perceval_path path to perceval binary
-#' @param bugzilla_site link to specific bugzilla site
-#' @param datetime fetch bugs updated since this date (in any ISO 8601 format, e.g., 'YYYY-MM-DD HH:mm:SS+|-HH:MM'))
-#' @param save_file_path the file path, name and extension (should be .json) to save the file.
-#' @param max_bugs the maximum number of bugs requested on the same query. This acts as the limit parameter
-#' in the Bugzilla REST API. Bugzilla sites may have specific limits set, so make sure to change the max_bugs
-#' parameter accordingly to correctly download the data when using the "bugzillarest" backend.
-#' @seealso \code{\link{parse_bugzilla_perceval_rest_issue_comments}} a parser function to parse bugzilla data
-#' @return json object with bugzilla data
-#' @export
-download_bugzilla_perceval_rest_issue_comments <- function(perceval_path, bugzilla_site, datetime, save_file_path, max_bugs=500){
-  perceval_path <- path.expand(perceval_path)
-  save_file_path <- path.expand(save_file_path)
-  json_data <- system2(perceval_path,
-                       args = c('bugzillarest', bugzilla_site, '--json-line',
-                                '--from-date', paste0('"',datetime,'"'),
-                                "--output",save_file_path,
-                                "--max-bugs", max_bugs),
-                       stdout = TRUE,
-                       stderr = FALSE)
-  return(save_file_path)
-}
-
 #' Download project data (issues and comments) from bugzilla site
 #' Note: The first comment in every issue is the issue description
 #' @param bugzilla_site URL to specific bugzilla site
 #' @param start_timestamp when to start bug retrieval (ex. 2023-01-01T00:14:57Z)
 #' @param save_folder_path the full *folder* path where the bugzilla issues will be stored
+#' @param project_key the project key of the project which can be found in the respective config file
 #' @param limit_upperbound the number of issues saved in each page file. Some bugzilla sites have limits set on how many bugs
 #' can be retrieved in one GET request, in which case, the limit set by the bugzilla site will be used in place of
 #' limit_upperbound to ensure full bug retrieval.
+#' @param comments set true to download issues with comments, leave as false to download only issues (without comments)
+#' @param verbose set true to print execution details
 #' @seealso \code{\link{parse_bugzilla_rest_issues_comments}} a parser function to parse Bugzilla issues and comments data
 #' @export
-download_bugzilla_rest_issues_comments <- function(bugzilla_site, start_timestamp, save_folder_path, limit_upperbound = 500) {
+download_bugzilla_rest_issues_comments <- function(bugzilla_site,
```

Comment:
Rename function to download_bugzilla_rest_issues_comments_by_date.

Reply:
Create a new function called download_bugzilla_rest_issues_comments. This function should have the exact code logic as this one except it will have a `query` parameter.

Then, the download_bugzilla_rest_issues_comments_by_date calls the function (download_bugzilla_rest_issues_comments) specifying the query parameter to be

`?creation_time=", start_timestamp, "&include_fields=_default,comments", "&limit=", limit, "&offset=", offset` if comments = TRUE, 

or else 
`?creation_time=", start_timestamp, "&limit=", limit, "&offset=", offset` if comments = FALSE.

Reply:
In the refresh function, call the by date function here.

Reply:
The logic for the naming convention of the files is in the download_bugzilla_rest_issues_comments.